switchdev_utils/config-service/README.md

```diff
@@ -0,0 +1,64 @@
+# Switchdev Network Configuration Service 
+## Description
+This folder contains SR-IOV, OVS and network configuration service for switchdev and VF-LAG  
+
+The configuration listed below shall be performed as part of a systemd script triggered during machine boot and IFCFG files for network configuration and each configuration step is optionally performed on specified interfaces via configuration file:  
+- Create VFs  
+- Move to switchdev  
+- Set VF trust for all VFs  
+- Create bond over PFs  
+- Create OVS bridges  
+- Enable hw-offload in OVS  
+- Connect bond or sriov pf to bridge  
+- Create sriov_config service
+- Create sriov_bond service
+
+## Prerequisites
+The below is assumed to be installed/configured on the host
+- NICs Firmware up to date (as bundled with MLNX_OFED)
+- (MLNX_OFED 5.2 or above ) or (mft 4.16 or above)
+- NICs NVCONFIG configures SR-IOV to support VFs per PF
+  - Mlxconfig -d <device> set SRIOV_EN=true NUM_OF_VFS=<numOfVfs>
+    - Note: reboot required to apply SRIOV firmware configuration
+- openvswitch
+
+## Config yaml file  
+The switchdev configuration service config file is a yaml file consists of the following sections:  
+- hwOffloadEnabled:  
+Configure OVS to enable or disable hardware offload  
+- pfs:  
+List of SRIOV PF objects which need to be configured, and has the following attributes:  
+    - name: The name of the sriov pf interface.  
+    - numOfVfs: Number of vfs to be configured on the sriov pf.  
+    - switchdev: Boolean, Enable switchdev mode.  
+    - vfTrust: Boolean, Configure vf trust.  
+- linuxBonds:  
+list of linux bond objects which need to be created and configured  
+    - name: The name of the bond interface  
+    - slaves: List of bond slaves interfaces (a subset of sriov pfs)  
+    - bondingOptions: Bonding options of the bond interface  
+- ovsBridges:  
+List of ovs bridges which need to be created and configured  
+    - name: The name of ovs bridge  
+    - ports: List of ovs bridge ports (a subset of sriov pfs or linux bonds)  
+
+## Install  
+Prepare you own config.yaml file with all the configuration you need and pass it to installation script or you can copy it directly to /etc/switchdev-config/config.yaml before execution  
+Install will do the following
+- create switchdev-config directory
+- copy switchdev_network_config to /etc/switchdev-config directory  
+
+Run the script  
+```$ bash ./install.sh [config.yaml]```  
+
+## Execution
```

Comment:
thats first time execution or on configuration change ? if so you should say this